packages/headless/src/app/navigatorContextProvider.ts

```diff
@@ -0,0 +1,14 @@
+export interface NavigatorContext {
+  referrer: string;
+  userAgent: string;
+  location: string;
+}
+
+export type NavigatorContextProvider = () => NavigatorContext;
+
+export const defaultBrowserNavigatorContextProvider: NavigatorContextProvider =
+  () => ({
+    referrer: document.referrer,
+    userAgent: navigator.userAgent,
+    location: window.location.href,
+  });
```

Comment:
An alternative implementation could be to use `relay.getMeta('')` here. The library is isomorphic, and will return `null` when used in a nodejs context. e.g.

```
const {referrer, userAgent, location} = relay.getMeta('')
```